components/gamification/store/StoreItemsList.tsx

```diff
@@ -19,15 +21,26 @@ const StoreItemsList  = ({ storeItemsPage, userItemsPage, isRefreshing, onRefres
   const [purchase, { isError: isPurchaseError, error: purchaseError }] = usePurchaseItemMutation();
   useError(purchaseError);
 
-  const { add } = useNotifications();
+  const [purchasingItemId, setPurchasingItemId] = useState('');
+
+  const { add, remove, clear } = useNotifications();
 
   const handleGemsPress = async (item: IStoreItemWithQuantity) => {
+    clear();
+
+    const purchasingNotificationId = add({ type: 'info', body: 'Purchasing...', time: 5000 });
+    setPurchasingItemId(item.id);
+
     const purchaseResponse = await purchase({ itemId: item.id });
     if (purchaseError || isPurchaseError || !isDataResponse(purchaseResponse)) {
+      setPurchasingItemId('');
       return;
     }
-    add({ type: 'success', body: 'Item bought successfully.' });
-    onRefresh();
+
+    remove(purchasingNotificationId);
+    setPurchasingItemId('');
```

Comment:
Also, this state is unnecessary, you should have handled this one level below, in the child. Set purchasing to true when "onGemPress" is triggered and set it to false when the "onGemPress" ends. If you do it this way, you rerender the whole list which may cause performance issues. If you do it the way I suggested, it will only rerender the card,